purge_olean.sh

```diff
@@ -1,13 +1,25 @@
-#! /bin/sh
-# olean=./src/tactic/ring.olean
-# lean_file=`echo $$olean_file | sed "s/\\.olean/.lean/"`
+#!/bin/sh
 
-for olean_file in `find . -name "*.olean"`;
+# Top-level root directory of the Git repository
+root=`git rev-parse --show-toplevel`
+
+# A list of directories containing .olean files. We are being conservative to
+# avoid traversing irrelevant directories and affecting directories we do not
+# want changed (e.g. $root/.git).
+dirs="$root/src $root/test"
+
+# Delete every <path>.olean without a matching <path>.lean.
+
+for olean_file in `find $dirs -name "*.olean"`
 do
-    echo olean file: $olean_file
     lean_file=`echo $olean_file | sed "s/\.olean/.lean/"`
```

Comment:
```suggestion
    lean_file=${olean_file/%.olean/.lean}
```

We don't need `sed`, `bash` can do the same (I hope other shells too)

Reply:
I'm ambivalent about which is better. Changed in de65ffb.